### ISSUE TITLE
FailedJobs filling with App\Jobs\TriggerAction "customer.enrich"

### ISSUE THREAD

**Issue by MarcoWel** [freescout-helpdesk/freescout#2348]
Thank you for implementing the [View Details] for Failed Jobs #2044 !

We now find the Failed Jobs list is consistently filling with App\Jobs\TriggerAction "customer.enrich". There are no details provided from which automated workflow this is coming from.

The details log always starts like this:
```
App\Jobs\TriggerAction Object
(
    [action] => customer.enrich
    [params] => Array
        (
            [0] => App\Customer Object
                (
                    [rememberCacheDriver] => array
                    [casts:protected] => Array
                        (
                            [meta] => array
                        )

                    [guarded:protected] => Array
                        (
                            [0] => id
                        )

                    [fillable:protected] => Array
                        (
                            [0] => first_name
                            [1] => last_name
                            [2] => company
                            [3] => job_title
                            [4] => address
                            [5] => city
                            [6] => state
                            [7] => zip
                            [8] => country
                            [9] => photo_url
                            [10] => age
                            [11] => gender
                            [12] => notes
                            [13] => channel
                            [14] => channel_id
                            [15] => social_profiles
                        )

...
```

The error log shows:
```
Illuminate\Queue\MaxAttemptsExceededException: App\Jobs\TriggerAction has been attempted too many times or run too long. The job may have previously timed out. in /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php:394
Stack trace:
#0 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(314): Illuminate\Queue\Worker->markJobAsFailedIfAlreadyExceedsMaxAttempts()
#1 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(270): Illuminate\Queue\Worker->process()
#2 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Worker.php(114): Illuminate\Queue\Worker->runJob()
#3 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(101): Illuminate\Queue\Worker->daemon()
#4 /var/www/html/vendor/laravel/framework/src/Illuminate/Queue/Console/WorkCommand.php(85): Illuminate\Queue\Console\WorkCommand->runWorker()
#5 [internal function]: Illuminate\Queue\Console\WorkCommand->handle()
#6 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(29): call_user_func_array()
#7 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(87): Illuminate\Container\BoundMethod::Illuminate\Container\{closure}()
#8 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/BoundMethod.php(31): Illuminate\Container\BoundMethod::callBoundMethod()
#9 /var/www/html/overrides/laravel/framework/src/Illuminate/Container/Container.php(549): Illuminate\Container\BoundMethod::call()
#10 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(183): Illuminate\Container\Container->call()
#11 /var/www/html/vendor/symfony/console/Command/Command.php(255): Illuminate\Console\Command->execute()
#12 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Command.php(170): Symfony\Component\Console\Command\Command->run()
#13 /var/www/html/vendor/symfony/console/Application.php(992): Illuminate\Console\Command->run()
#14 /var/www/html/vendor/symfony/console/Application.php(255): Symfony\Component\Console\Application->doRunCommand()
#15 /var/www/html/vendor/symfony/console/Application.php(148): Symfony\Component\Console\Application->doRun()
#16 /var/www/html/vendor/laravel/framework/src/Illuminate/Console/Application.php(88): Symfony\Component\Console\Application->run()
#17 /var/www/html/vendor/laravel/framework/src/Illuminate/Foundation/Console/Kernel.php(121): Illuminate\Console\Application->run()
#18 /var/www/html/artisan(62): Illuminate\Foundation\Console\Kernel->handle()
#19 {main}
```

Any ideas?

**Comment by freescout-helpdesk** [comment]
It's from https://freescout.net/module/customer-data-enrichment/. Just ignore them, as customer enrichment if performed in the background and sometimes customer enrichment services are not responding.

**Comment by MarcoWel** [comment]
Thank you for the quick reply. Is this module fixable in a way it doesn't permanently fill the Failed Job Queue? It happens since we use that module. I deactivated it for now.

**Comment by freescout-helpdesk** [comment]
Fixed in Customer Data Enrichment v1.0.2
